src/withForm.ts

```diff
@@ -0,0 +1,192 @@
+        this.validateOnChange ??
+        true
+
+      // isLiteForm atribute for HOCs can find their form element
+      this.setAttribute(IS_LITE_FORM, 'true')
+
+      this.handleReset()
+      super.connectedCallback && super.connectedCallback()
+    }
+
+    get _values(): ValueMap {
+      return this.values
+    }
+
+    set _values(values: ValueMap) {
+      this.values = values
+
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.valuesChange, {
+          detail: { values: this.values }
+        })
+      )
+    }
+
+    get _errors(): ErrorMap {
+      return this.errors
+    }
+
+    set _errors(errors: ErrorMap) {
+      this.errors = errors
+
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.errorsChange, {
+          detail: { errors: this.errors, touched: this.touched }
+        })
+      )
+    }
+
+    get _touched(): TouchedMap {
+      return this.touched
+    }
+
+    set _touched(touched: TouchedMap) {
+      this.touched = touched
+      this.dispatchEvent(
+        new CustomEvent(EVENTS.errorsChange, {
+          detail: { errors: this.errors, touched: this.touched }
+        })
+      )
+    }
+
+    handleReset = (e?:Event): void => {
+      e && e.preventDefault()
+      this._values = cloneDeep(this._initialValues)
+      this._touched = {}
+      this._errors = {}
+      this.isValid = undefined
+    }
+
+    handleSubmit = (e?:Event): void => {
+      e && e.preventDefault()
+
+      this.handleValidate()
+
+      if (this.isValid) {
+        this._onSubmit(this._values, this)
+      } else {
+        // Touch all fields to show validation errors
+        this._touched = Object.keys(this._validationSchema).reduce(
+          (obj: { [propName: string]: boolean; }, fieldName: string) => {
+            set(obj, fieldName, true)
+            return obj
+          },
+          {}
+        )
+      }
+    }
+
+    // Set touched and handleValidate if needed
+    setTouched = (name: string): void => {
+      this._touched = set(this._touched, name, true)
+      if (this._validateOnBlur) {
+        this.handleValidate()
+      }     
+    }
+
+    // Handle Blur events
+    handleBlur = (event: Event | CustomEvent): void => {
+      const eventTarget = getEventTarget(event)
+      if (eventTarget) {
+        this.setTouched(eventTarget.name || eventTarget.id);
+      }
+    }
+
+    // Set the value for the given name.
+    // Triggers optional validation when validate is true.
+    setValue = (name: string, value:InputValue, validate = false): void => {
+      this._values = set(cloneDeep(this.values), name, value)
+      if (validate && this._validateOnChange) {
+        this.handleValidate()
+      }
+    }
+
+    // handle Change events
+    handleChange = (event: Event | CustomEvent): void  => {
+      const eventTarget = getEventTarget(event)
+      if (eventTarget) {
+        const value = getValueFromEventTarget(eventTarget)
+        this.setValue(eventTarget.name || eventTarget.id, value, true)
```

Comment:
maybe it will be better: 
`this.setValue(eventTarget.name || eventTarget.id, value, this._validateOnChange)`
and remove `this._validateOnChange` from condition in withForm.setValue

Reply:
Done.
Your suggestions to pass `this._validateOnChange` and `this._validateOnBlur` make complete sense.